Source/IntegrationTests/TestFiles/LitTests/LitTest/reveal/revealFunctions.dfy

```diff
@@ -0,0 +1,84 @@
+// RUN: ! %verify --type-system-refresh --allow-axioms --bprint:%t.bpl --isolate-assertions --boogie "/printPruned:%S/pruned" %s > %t
+// RUN: %diff "%s.expect" "%t"
+
+function P(x: int): bool {
+  true
+}
+
+method HideAndRevealHappyFlow() {
+  hide *;
+  
+  if (*) {
+    reveal P;
+    assert P(0);
+  } else {
+    assert P(0); // error
+  }
+  reveal P;
+  if (*) {
+    assert P(1);
+  } else {
+    hide *;
+    assert P(1); // error
+  }
+}
+
+function EnsuresFuncFoo(): bool
+  ensures EnsuresFuncFoo()
+
+method UseEnsuresFuncFoo() {
+  hide *;
+  if (*) {
+    reveal EnsuresFuncFoo;
+    assert EnsuresFuncFoo();
+  } else {
+    assert EnsuresFuncFoo(); // no error. Could be changed in the future
```

Comment:
Yeah I would find it weird that the ensures would not be hidden if we have hide *.